roles/openshift_monitor_availability/README.md

```diff
@@ -0,0 +1,34 @@
+# OpenShift Availability Monitoring
+
+### **NOTE: This component is unsupported in OCP at this time.**
+
+These are black box tests of several components which elable SLA verification by
+capturing metrics for SLIs. All of the tests in this role should be used to
+measure availability. Each test should expose metrics endpoints for scraping by
+the monitoring platform (Prometheus).
+
+All of the test applications are installed into the `openshift-monitor-availability` namespace and are enabled/disabled using the `openshift_monitor_availability_install` variable.
+
+## Adding a new application
+
+To add a new application to the installer:
+
+1. Add an OpenShift Template to the `files` directory which can be used with `oc process | oc apply` to install the application.
+2. Create an Ansible task in the `tasks` directory, e.g. `install_{APP}.yaml`. The task should install the application into the `openshift-monitor-availability` namespace.
+3. Include the new task in `install.yaml`:
+
+        - import_tasks: install_{APP}.yaml
+
+
+## Guidelines
+
+Here are some guidelines for applications:
+
+* App metrics endpoints **must be secured**. Use the [oauth-proxy](https://github.com/openshift/oauth-proxy) or [kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy).
```

Comment:
Was that just a general expression of incredulity or was there a more specific concern? lol

Reply:
it's a concern.  it sounds burdensome and not sure why it's necessary as a requirement?

we aren't even requiring https, so....

Reply:
Just to make sure we aren't talking past each other... are you concerned about a requirement for the **application itself** to serve HTTPS, or are you concerned about the requirement to **front your application with an HTTPS proxy**? The latter is the intent (your app listens HTTP on loopback, and the HTTPS proxy is a boilerplate configuration detail in the template within this role).

Reply:
i'm concerned about the auth requirement, but if it's just an additional pod/service defined in the template then i guess my concerns are alleviated.

Reply:
Sorry, I left out the "why"... the platform metrics endpoints must be trusted sources of truth- if we permitted unauthenticated scraping, that opens a hole for a rogue service to get itself scraped by Prometheus (assuming it managed to get itself discovered).

Reply:
that would explain why it requires https, not sure it explains why auth is required?

Reply:
@brancz @derekwaynecarr would I need to add a SAR here to actually implement authz as I have described it? With just a certificate check via rbac-proxy like I have here, couldn't any service using the service signing certs get scraped if they end up matched (how a service we don't want to get matched makes itself match our scrape selectors seems like another topc)?

Reply:
not even sure how rbac would help.  a rogue endpoint would just say "yup, you're allowed" to all requests.

the only way that's obvious to me to keep a rogue endpoint from being scraped (assuming they can trick prometheus into trying to scrape it) is to require prometheus to only trust a very specific CA where only our metric endpoints have/can get certificates signed by that CA.

as soon as prometheus trusts the cluster CA, anyone who can create services on the cluster can presumably create something that prometheus will scrape.

Reply:
Yeah, would love for others to chime in here... right now authz seems implicit in the fact that we configure Prom to only scrape from ServiceMonitors in a trusted namespace, pointing at services in trusted namespaces (i.e. only we can create services in those namespaces)...

Reply:
I thought one of the reasons for requiring auth for metrics is to make sure random unprivileged  users on the cluster won't be able to scrape metrics of sensitive cluster components

Reply:
Got some education from @brancz, let's see if this helps clarify:

* The security intent here is to establish trust from endpoint->requester (prometheus), **not** to establish mutual trust.
* TLS for transport security to prevent outright eavesdropping on the pod network.
* SAR for authorization to establish endpoint->requester trust (with the caveat that this is sort of fuzzy, as the RBAC-based SAR can't precisely identify "prometheus" except implicitly via capabilities checks).

If all that's accurate, there should be a SAR here and the docs need fixed.